### PR TITLE
Add support for reacting to dark/light mode theme changes with winit

### PR DESCRIPTION
**Pull request by tronical**
On macOS and Windows, use winit's `Window::theme()` instead of dark-light, to detect the theme.

Always react to WindowEvent::ThemeChange. It's delivered on macOS, Windows, and the web.

### CODE REVIEW

File: internal/renderers/skia/Cargo.toml [hunk 1]
```diff
@@ -49,7 +49,7 @@ glutin = { version = "0.30", default-features = false, features = ["egl", "wgl"]
 winit = { version = "0.28.1", optional = true, default-features = false }
 
 [target.'cfg(target_family = "windows")'.dependencies]
-winapi = { version = "0.3", features = ["dwrite", "d3d12", "dxgi", "dxgi1_2", "dxgi1_3", "dxgi1_4", "d3d12sdklayers", "synchapi", "winbase"] }
+winapi = { version = "0.3", features = ["impl-default", "dwrite", "d3d12", "dxgi", "dxgi1_2", "dxgi1_3", "dxgi1_4", "d3d12sdklayers", "synchapi", "winbase"] }
```
Comment (ogoffart) [inline internal/renderers/skia/Cargo.toml:52]: Is that related?

**Review by ogoffart — APPROVED** [review]

File: internal/renderers/skia/Cargo.toml (hunk 1 above)
Comment (tronical) [reply internal/renderers/skia/Cargo.toml:52]: Yes. Previously this feature was implicitly selected (through dark-light-mode). Without that dependency we need to select it explicitly.
